src/ngraph/op/embedding_segments_sum.hpp

```diff
@@ -0,0 +1,92 @@
+                ///
+                /// EmbeddingSegmentsSum constructs an output tensor by replacing every index in a
+                /// given
+                /// input tensor with a row (from the weights matrix) at that index
+                ///
+                /// \param 'emb_table' tensor containing the embedding lookup table of the module of
+                /// shape [num_emb, emb_dim1, emb_dim2, ...] and  of type T
+                /// \param 'indices' tensor of shape [num_indices] and of type T_IND. Required
+                /// \param `segment_ids` tensor of shape `[num_indices]` and of type *T_IND* with
+                /// indices
+                /// into the output Tensor. Values should be sorted and can be repeated. Required.
+                /// \param `num_segments` scalar of type *T_IND* indicating the number of segments.
+                /// Required.
+                /// \param 'per_sample_weights' tensor of the same shape as indices and of type T.
+                /// Each value in this tensor are multiplied with each
+                /// value pooled from embedding table for each index. Optional.
+                /// \param 'default_index' scalar of type T_IND containing default index in
+                /// embedding
+                /// table to fill empty "bags". If not provided empty "bags"
+                /// are filled with zeros. Optional.
+
+                EmbeddingSegmentsSum(const Output<Node>& emb_table,
+                                     const Output<Node>& indices,
+                                     const Output<Node>& segment_ids,
+                                     const Output<Node>& num_segments,
+                                     const Output<Node>& per_sample_weights,
+                                     const Output<Node>& default_index);
+
+                EmbeddingSegmentsSum(const Output<Node>& emb_table,
+                                     const Output<Node>& indices,
+                                     const Output<Node>& segment_ids,
+                                     const Output<Node>& num_segments,
+                                     const Output<Node>& per_sample_weights);
+
+                EmbeddingSegmentsSum(const Output<Node>& emb_table,
+                                     const Output<Node>& indices,
+                                     const Output<Node>& segment_ids,
+                                     const Output<Node>& num_segments);
+
+                void validate_and_infer_types() override;
+
+                virtual std::shared_ptr<Node>
+                    clone_with_new_inputs(const OutputVector& new_args) const override;
+
+            private:
+                static constexpr int EMB_TABLE = 0;
+                static constexpr int INDICES = 1;
+                static constexpr int SEGMENT_IDS = 2;
+                static constexpr int NUM_SEGMENTS = 3;
+                static constexpr int PER_SAMPLE_WEIGHTS = 4;
```

Comment:
Please, pay attention to a change in an order of inputs described here https://gitlab-icv.inn.intel.com/inference-engine/dldt/merge_requests/7490